subsys/nrf_security/src/drivers/cracen/cracenpsa/src/sign.c

```diff
@@ -340,69 +415,65 @@ static psa_status_t cracen_signature_ecc_verify(bool is_message,
 						size_t input_length, const uint8_t *signature,
 						size_t signature_length)
 {
-	int si_status = 0;
-	const struct sx_pk_ecurve *curve;
-	struct si_sig_pubkey pubkey = {0};
-	struct si_sig_signature sign = {0};
-	char pubkey_buffer[132] = {0}; /* 521 bits * 2 */
+	psa_key_type_t key_type = psa_get_key_type(attributes);
 
-	/* Workmem for sicrypto ecc verify task is digest size. */
-	char workmem[PSA_HASH_MAX_SIZE];
-	struct sitask t;
+	psa_status_t status = validate_ec_signature_inputs(is_message, attributes, key_type, alg);
 
-	if (!PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(psa_get_key_type(attributes)) &&
-	    !PSA_KEY_TYPE_IS_ECC_KEY_PAIR(psa_get_key_type(attributes))) {
-		return PSA_ERROR_NOT_SUPPORTED;
+	if (status != PSA_SUCCESS) {
+		return status;
 	}
 
-	if (!PSA_ALG_IS_ECDSA(alg) && alg != PSA_ALG_PURE_EDDSA && !PSA_ALG_IS_HASH_EDDSA(alg)) {
+	size_t public_key_size =
+		PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(key_type, psa_get_key_bits(attributes));
+	if (public_key_size == 0) {
 		return PSA_ERROR_NOT_SUPPORTED;
 	}
 
-	if (((is_message) && (!SI_PSA_IS_KEY_FLAG(PSA_KEY_USAGE_VERIFY_MESSAGE, attributes))) ||
-	    ((!is_message) && (!SI_PSA_IS_KEY_FLAG(PSA_KEY_USAGE_VERIFY_HASH, attributes)))) {
-		return PSA_ERROR_INVALID_ARGUMENT;
-	}
+	char pubkey_buffer[public_key_size];
+	const struct sx_pk_ecurve *curve = NULL;
 
-	si_task_init(&t, workmem, sizeof(workmem));
-	si_status = cracen_signature_prepare_ec_pubkey(&t, &pubkey, (char *)key_buffer,
-						       key_buffer_size, &curve, alg, attributes,
-						       is_message, input_length, pubkey_buffer);
-	if (si_status) {
+	int si_status = cracen_signature_prepare_ec_pubkey(
+		(const char *)key_buffer, key_buffer_size, &curve, alg, attributes, pubkey_buffer);
+	if (si_status != PSA_SUCCESS) {
```

Comment:
Nit nit but this doesn't return a PSA error code. Right now there is no difference since PSA_SUCCESS and SX_OK are both zero. If you do other changes you can do it otherwise it can be fixed later.